resolving/resolver-base/src/index.ts

```diff
@@ -38,21 +38,69 @@ export type Resolution =
   GitResolution |
   ({ type: string } & object)
 
-export interface ResolveResult {
+export type ResolveVia =
+  | 'npm-registry'
+  | 'jsr-registry'
+  | 'git-repository'
+  | 'local-filesystem'
+  | 'url'
+  | 'workspace'
+
+export type ResolveResult =
+  | NpmResolveResult
+  | JsrResolveResult
+  | GitResolveResult
+  | LocalResolveResult
+  | UrlResolveResult
+  | WorkspaceResolveResult
+
+export interface ResolveResultBase {
   id: PkgResolutionId
   latest?: string
   publishedAt?: string
   manifest?: DependencyManifest
   resolution: Resolution
-  resolvedVia: 'npm-registry' | 'git-repository' | 'local-filesystem' | 'workspace' | 'url' | string
+  resolvedVia: ResolveVia
   normalizedBareSpecifier?: string
   alias?: string
 }
 
+export interface NpmResolveResult extends ResolveResultBase {
+  latest: string
+  manifest: DependencyManifest
```

Comment:
`npm-resolver` actually returns `PackageInRegistry` here.

Reply:
That's the manifest from the package document (aka pacument)